expression/builtin_time.go

```diff
@@ -1728,6 +1732,32 @@ func evalFromUnixTime(ctx sessionctx.Context, fsp int, unixTimeStamp *types.MyDe
 	return t, false, nil
 }
 
+// fieldString returns true if precision cannot be determined
+func fieldString(fieldType byte) bool {
+	switch fieldType {
+	case mysql.TypeString, mysql.TypeVarchar, mysql.TypeTinyBlob,
+		mysql.TypeMediumBlob, mysql.TypeLongBlob, mysql.TypeBlob:
+		return true
+	default:
+		return false
+	}
+}
+
+// Improve Unix Time 6-bit precision
+func updateFromUnixTimePrecision(expr Expression) {
+	x, ok := (expr).(*ScalarFunction)
+	if ok {
+		// FixUnixtimePrecision is used to adjust FromUnixTime precision #Fixbug35184
+		if x.FuncName.L == ast.Cast {
+			if x.RetType.GetDecimal() == 0 {
```

Comment:
1. why need to check `x.RetType.GetDecimal()`, sometime the decimal can be -1, and it also need to be updated.
2. why need check if it is cast function? I think just check if the return type if decimal is ok

Reply:
1）string type must be equal to 0
2）The current application scenario is cast